src/internal_events/template.rs

```diff
@@ -0,0 +1,30 @@
+use super::InternalEvent;
+use crate::template::TemplateRenderingError;
+use metrics::counter;
+
+pub struct TemplateRenderingFailed<'a> {
+    pub field: Option<&'a str>,
+    pub drop_event: bool,
+    pub error: TemplateRenderingError,
+}
+
+impl<'a> InternalEvent for TemplateRenderingFailed<'a> {
+    fn emit_logs(&self) {
+        let mut msg = "Failed to render template".to_owned();
+        if let Some(field) = self.field {
+            msg.push_str(&format!(" for \"{}\"", field));
+        }
+        if self.drop_event {
+            msg.push_str("; discarding event");
+        }
+        msg.push('.');
+        warn!(message = %msg, error = %self.error, internal_log_rate_secs = 30);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("template_rendering_errors_total", 1);
```

Comment:
I'm actually wondering if this shouldn't be a `processing_errors_total` with a `"error_type" => "template_render_failed"`.

Additionally, we might want to label the counter with the field that failed to render so that it is easier to figure out which fields have template render errors for components that support templates for multiple options.

Reply:
I've asked this question in discord, but I think no one has replied. I'm also in favor of `processing_errors_total` here.

Reply:
Just adding a reminder to change the cue docs accordingly after that.

Reply:
I changed to `processing_errors_total`, using existing error type `render_error`.